File: CPP/Factorial numbers.cpp
using namespace std;
vector<long long> helper(long long n, long long i, long long fact, vector<long long> res){
    if(fact <= n){
        res.push_back(fact);
        return helper(n, i+1, fact * (i+1), res);
    }
    return res;



}
vector<long long> factorialNumbers(long long n) {
    vector<long long > res;
    return helper(n, 1, 1, res);
}

File: Java/Factorial_numbers.java
import java.util.ArrayList;
import java.util.List;

public class Solution {
    public static List<Long> helper(long n, long i , long fact, List<Long> res){
        if(fact <= n){
            res.add(fact);
            helper(n, i+1, fact*(i+1), res);
        }
        return res;
    }
    public static List<Long> factorialNumbers(long n) {
        List<Long> res = new ArrayList<>();
        return helper(n, 1, 1, res);
    }
}
